packages/cardano-services/test/ChainHistory/DbSyncChainHistoryProvider/ChainHistoryBuilder.test.ts

```diff
@@ -0,0 +1,191 @@
+    });
+    test('query transaction withdrawals with empty array', async () => {
+      const result = await builder.queryWithdrawalsByHashes([]);
+      expect(result.size).toEqual(0);
+    });
+    test('query transaction withdrawals when tx not found or has no withdrawals', async () => {
+      const result = await builder.queryWithdrawalsByHashes([
+        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000'),
+        Cardano.TransactionId('cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819')
+      ]);
+      expect(result.size).toEqual(0);
+    });
+  });
+
+  describe('queryRedeemersByHashes', () => {
+    test('query transaction redeemers by tx hashes', async () => {
+      const result = await builder.queryRedeemersByHashes([
+        Cardano.TransactionId('5acd6efb1b66299f1c5a2c4221af4bcaa4ba9929e8e6aa0e3f48707fa1796fc3'),
+        Cardano.TransactionId('24e75c64a309fd8fb400933795b2522ca818cba80a3838c2ff14cec2cc8ffe4e')
+      ]);
+      expect(result.size).toEqual(2);
+      expect(result).toMatchSnapshot();
+    });
+    test('query transaction redeemers with empty array', async () => {
+      const result = await builder.queryRedeemersByHashes([]);
+      expect(result.size).toEqual(0);
+    });
+    test('query transaction redeemers when tx not found or has no redeemers', async () => {
+      const result = await builder.queryRedeemersByHashes([
+        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000'),
+        Cardano.TransactionId('cefd2fcf657e5e5d6c35975f4e052f427819391b153ebb16ad8aa107ba5a3819')
+      ]);
+      expect(result.size).toEqual(0);
+    });
+  });
+
+  describe('queryTxMetadataByHashes', () => {
+    test('query transaction metadata by tx hashes', async () => {
+      const result = await builder.queryTxMetadataByHashes([
+        Cardano.TransactionId('3d2278e9cef71c79720a11bc3e08acbbd5f2175f7015d358c867fc9b419ae0b2'),
+        Cardano.TransactionId('545c4656544054045f5a4db0e962f6b09fc6d98b0303d42f3f006e3d920d3720')
+      ]);
+      expect(result.size).toEqual(2);
+      expect(result).toMatchSnapshot();
+    });
+    test('query transaction metadata with empty array', async () => {
+      const result = await builder.queryTxMetadataByHashes([]);
+      expect(result.size).toEqual(0);
+    });
+    test('query transaction metadata when tx not found or has no metadata', async () => {
+      const result = await builder.queryTxMetadataByHashes([
+        Cardano.TransactionId('0000000000000000000000000000000000000000000000000000000000000000')
+      ]);
+      expect(result.size).toEqual(0);
+    });
+  });
+
+  describe('queryProtocolParams', () => {
+    test('query protocol parameters from last epoch', async () => {
+      const result = await builder.queryProtocolParams();
+      expect(result).toMatchSnapshot();
+    });
+  });
+
+  describe('queryCertificatesByHashes', () => {
```

Comment:
missing cases for:
-    MirCertificate
-    GenesisKeyDelegationCertificate  
-    StakeKeyDeregistrationCertificate

Reply:
I'm not sure if we can get `GenesisKeyDelegationCertificate` from db-sync, I didn't make any queries to get them and they are not being fetched by the blockfrost provider either.

Added the others in https://github.com/input-output-hk/cardano-js-sdk/pull/243/commits/5372e875ce7ce0f93277b75e1f5b2bdbaa008274

Reply:
Please leave a comment in the test to explain why there are no tests for these cert types @lgobbi-atix

Reply:
It's blocked by https://github.com/input-output-hk/cardano-db-sync/issues/290

Reply:
https://github.com/input-output-hk/cardano-js-sdk/commit/07a978ee39e38f0f8a62a662895af610b22db10c#diff-3e16e6fa3a0c30bc7dc482538653e7edca22402ca67614ba46757d93f533b0e8R150